include/okapi/impl/device/opticalSensor.hpp

```diff
@@ -0,0 +1,118 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+#pragma once
+
+#include "api.h"
+#include "okapi/api/control/controllerInput.hpp"
+#include "okapi/api/filter/passthroughFilter.hpp"
+#include <memory>
+
+namespace okapi {
+class OpticalSensor : public ControllerInput<double> {
+  public:
+  /**
+   * An optical sensor on a V5 port.
+   *
+   * ```cpp
+   * auto os = OpticalSensor(1);
+   * ```
+   *
+   * @param iport The V5 port the device uses
+   */
+  OpticalSensor(std::uint8_t iport);
+
+  virtual ~OpticalSensor();
+
+  /**
+   * Returns the current sensor's hue value (0-359.99).
+   *
+   * @return current value
+   */
+  virtual double get();
+
+  /**
+   * Returns the current sensor's brightness value (0-1.0).
+   *
+   * @return current value
+   */
+  virtual double get_brightness();
+
+  /**
+   * Returns the current sensor's saturation value (0-1.0).
+   *
+   * @return current value
+   */
+  virtual double get_saturation();
+
+  /**
+   * Returns the current sensor's LED brightness
+   *
+   * value that ranges from 0 to 100
+   *
+   * @return current value
+   */
+  virtual int32_t get_led_brightness();
+
+  /**
+   * Set the pwm value of the White LED on the sensor
+   *
+   * value that ranges from 0 to 100
+   */
+  virtual int32_t set_led_brightness(int32_t v);
```

Comment:
As per okapi convention, all the parameters need the `i` prefix in their name. What is `v`? Should it be `ipwm`?
Maybe the function should be renamed to `set_led_pwm`  to stay consistent with `get_led_pwm` and the PROS API.